File: leetcode/algorithms/c++/Linked List Cycle/Linked List Cycle.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution
{
public:
    bool hasCycle(ListNode *head)
    {
        int LIMIT = 1e6;
        int i = 0;
        while (head != NULL && i < LIMIT)
        {
            head = head->next;
            ++i;
        }
        return head != NULL;
    }
};

File: leetcode/algorithms/c#/Linked List Cycle/Linked List Cycle.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution
{
    public bool HasCycle(ListNode head)
    {

        ListNode slow = head;
        ListNode fast = head;
        while (slow != null && slow.next != null && fast != null && fast.next != null && fast.next.next != null)
        {
            slow = slow.next;
            fast = fast.next.next;
            //Console.WriteLine(string.Format("{0}, {1}", slow.val, fast.val));
            if (slow == fast)
                return true;

        }
        return false;
    }
}